### PR TITLE
Upgrade for Debug Tools for Slots and Plugins

### PR DESCRIPTION
**Pull request by okbel**
## Upgrade for Debug Tools for Slots and Plugins

- `plugin_config` changed to `plugins_config`
- Adds a docs for Slots (in Spanish)
- Passing `debug:true` to the `plugins_config` highlights the slots and when you hover them you get a tooltip with the name of the slot. Simple, interactive and helpful tool to see which slots are available.

```js
plugins_config: {
  debug: true
}
```

Outputs: 
out2

### CODE REVIEW

**Review by kgardnr — DISMISSED** [review]
Just need to add the deprecation warning and keep supporting old and new - as discussed in scrum. Thanks @okbel!

File: client/coral-framework/services/plugins.js [hunk 1]
```diff
@@ -83,11 +83,26 @@ class PluginsService {
    * getSlotComponentProps calculate the props we would pass to the slot component.
    * query datas are only passed to the component if it is defined in `component.fragments`.
    */
+  showPluginsConfigWarning = true;
```
Comment (alinhle) [inline client/coral-framework/services/plugins.js]: You can use https://lodash.com/docs/4.17.5#once instead

File: client/coral-framework/services/plugins.js [hunk 2]
```diff
@@ -83,11 +83,26 @@ class PluginsService {
...
+
   getSlotComponentProps(component, reduxState, props, queryData) {
-    const pluginConfig = get(reduxState, 'config.plugin_config') || emptyConfig;
+    if (
+      !!get(reduxState, 'config.plugin_config') &&
+      this.showPluginsConfigWarning
+    ) {
+      console.warn(
```
Comment (alinhle) [inline client/coral-framework/services/plugins.js]: Hide warnings when in production.

File: client/coral-embed-stream/src/tabs/stream/containers/Stream.js [hunk 3]
```diff
@@ -465,7 +465,7 @@ const mapStateToProps = state => ({
   activeStreamTab: state.stream.activeTab,
   previousStreamTab: state.stream.previousTab,
   commentClassNames: state.stream.commentClassNames,
-  pluginConfig: state.config.plugin_config,
+  pluginsConfig: state.config.plugins_config || state.config.plugin_config,
```
Comment (alinhle) [inline client/coral-embed-stream/src/tabs/stream/containers/Stream.js]: Add `@Deprecated` for the old one, or refactor into a selector

File: plugin-api/beta/client/selectors/index.js [hunk 4]
```diff
@@ -1 +1 @@
-export const pluginConfigSelector = state => state.config.pluginConfig;
+export const pluginsConfigSelector = state => state.config.pluginsConfig;
```
Comment (alinhle) [inline plugin-api/beta/client/selectors/index.js]: Should support both, also add `@Deprecate` comment

File: client/coral-embed-stream/src/tabs/stream/containers/Stream.js (hunk 3 above)
Comment (alinhle) [reply client/coral-embed-stream/src/tabs/stream/containers/Stream.js]: So we can easily find it.

File: client/coral-framework/services/plugins.js [hunk 5]
```diff
@@ -83,11 +83,26 @@ class PluginsService {
    * getSlotComponentProps calculate the props we would pass to the slot component.
    * query datas are only passed to the component if it is defined in `component.fragments`.
    */
+  showPluginsConfigWarning = true;
+
   getSlotComponentProps(component, reduxState, props, queryData) {
-    const pluginConfig = get(reduxState, 'config.plugin_config') || emptyConfig;
+    if (
```
Comment (alinhle) [inline client/coral-framework/services/plugins.js]: Should add `@Deprecate` comment so we can easily discover it again.

**Review by alinhle — CHANGES_REQUESTED** [review]
Small tweaks

**Comment by okbel** [comment]
Ready to be tested!

File: client/coral-embed-stream/src/reducers/index.js [hunk 6]
```diff
@@ -3,11 +3,13 @@ import embed from './embed';
...
 import profile from './profile';
+import debug from './debug';
 
 export default {
   login,
   embed,
   configure,
   stream,
   profile,
+  debug,
```
Comment (cvle) [inline client/coral-embed-stream/src/reducers/index.js]: Why do we need another reducer for `debug`. Don't we have a reducer in  `config` already with the `debug` value?

Comment (alinhle) [inline client/coral-embed-stream/src/reducers/index.js]: Why do we need another reducer for debug. Don't we have a reducer in config already with the debug value?

File: client/coral-embed-stream/src/tabs/stream/containers/Stream.js [hunk 7]
```diff
@@ -465,7 +465,9 @@ const mapStateToProps = state => ({
   activeStreamTab: state.stream.activeTab,
   previousStreamTab: state.stream.previousTab,
   commentClassNames: state.stream.commentClassNames,
-  pluginConfig: state.config.plugin_config,
+
+  // @Deprecated plugin_config
+  pluginsConfig: state.config.plugins_config || state.config.plugin_config,
```
Comment (alinhle) [inline client/coral-embed-stream/src/tabs/stream/containers/Stream.js]: Turns out we don't use this at all, do we?

File: client/coral-embed/src/StreamInterface.js [hunk 8]
```diff
@@ -3,10 +3,18 @@ export default class StreamInterface {
     this._stream = stream;
   }
 
+  dispatch(action) {
```
Comment (alinhle) [inline client/coral-embed/src/StreamInterface.js]: As discussed use a more specific method like `setDebug(boolean)` or similar.

File: client/coral-framework/components/Slot.js [hunk 9]
```diff
@@ -88,13 +95,24 @@ class Slot extends React.Component {
...
         className={cn(
-          { [styles.inline]: inline, [styles.debug]: pluginConfig.debug },
+          {
+            [styles.inline]: inline,
+            [styles.debug]: debug.plugins || pluginsConfig.debug,
+          },
           className,
           `talk-slot-${kebabCase(fill)}`
         )}
+        {...debugProps}
```
Comment (alinhle) [inline client/coral-framework/components/Slot.js]: Props passed to the plugins should be handled in the `plugins` service.

**Review by alinhle — CHANGES_REQUESTED** [review]

File: docs/source/04-06-slots-and-plugins.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+The first thing we should consider is what components will be affected by our plugin's functionality. For example, if we want to add functionality to all the comments that are rendered in a total list of comments, we would use the component `Comment`.
+
+The slots that are able to add functionality to comments start with `comment`, like `commentContent`, or `commentReactions`, as you can see above.
+
+### Disabling plugins via `plugins_config`
+
+Typically, you will manage plugins via your `plugins.json` file. If you want to remove a plugin, you would simply delete it there. However, we can also do this directly with the `plugins_config`.
+
+Let's look at our example article, `views/article.ejs`. Here we see can we have the Talk embed, and within the embed, we can also send a configuration object. To disable a plugin visually, we can pass `true` to the property `disable_components`. Like so:
```
Comment (alinhle) [inline docs/source/04-06-slots-and-plugins.md]: I guess you meant: `Here  we have the Talk embed, ...`

File: docs/source/plugins/slots-and-plugins.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
+---
```
Comment (alinhle) [inline docs/source/plugins/slots-and-plugins.md]: Duplicate file?

**Review by alinhle — CHANGES_REQUESTED** [review]

File: docs/source/04-06-slots-and-plugins.md (hunk 10 above)
Comment (kgardnr) [reply docs/source/04-06-slots-and-plugins.md]: "Here we can see we have the Talk embed..." is also correct - is it not clear?

File: docs/source/plugins/slots-and-plugins.md (hunk 11 above)
Comment (kgardnr) [reply docs/source/plugins/slots-and-plugins.md]: Yes will remove!

**Review by okbel — COMMENTED** [review]
😍

File: docs/source/04-06-slots-and-plugins.md (hunk 10 above)
Comment (alinhle) [reply docs/source/04-06-slots-and-plugins.md]: Is that correct English? I thought `that` would be required, e.g. `Here we can see that we have the Talk embed`

Comment (kgardnr) [reply docs/source/04-06-slots-and-plugins.md]: Oh lol, it's a typo, sorry.

Comment (kgardnr) [reply docs/source/04-06-slots-and-plugins.md]: Fixed!

**Review by alinhle — APPROVED** [review]

**Pull request merged**